lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1133,7 +1133,7 @@ export class UserAgentApplication {
       // login_hint
       else {
         // login_hint is account.userName
-        if (!qParams[SSOTypes.LOGIN_HINT]  && account.userName && !Utils.isEmpty(account.userName)) {
+        if (!qParams[SSOTypes.SID] && !qParams[SSOTypes.LOGIN_HINT] && account.userName && !Utils.isEmpty(account.userName)) {
```

Comment:
this looks find, I wonder if factoring this into variable will add to readability

```javascript
const thingWeAreCheckingFor = predicate && predicate && predicate; 
if(!thingWeAreCheckingFor) {  ... }
```

Reply:
no ... so this is a test branch, hence I haven't yet sent it for a review. This is for the customer to test the change. Any style that simplifies readability, can always be incorporated.